cmake/options.cmake

```diff
@@ -71,9 +71,10 @@ option(OSQUERY_NO_DEBUG_SYMBOLS "Whether to build without debug symbols or not,
 option(OSQUERY_BUILD_TESTS "Whether to enable and build tests or not")
 option(OSQUERY_BUILD_ROOT_TESTS "Whether to enable and build tests that require root access")
 
-if(DEFINED PLATFORM_LINUX)
+if(DEFINED PLATFORM_LINUX OR DEFINED PLATFORM_MACOS)
```

Comment:
This enables the subsequent options on macOS too, but for instance fuzzers won't be built anyway on macOS.
Also, I don't know how good it is the support for Valgrind?

I would suggest taking out just the `OSQUERY_ENABLE_ADDRESS_SANITIZER` and `OSQUERY_ENABLE_THREAD_SANITIZER` and enable these for macOS and Linux, and leave the rest on Linux only.

If then we want to enable fuzzers/valgrind on macOS too, I would do that on a different PR when we have tested this.